### PR TITLE
Fix #208: Add forgot password page

### LINKED ISSUE #208: [PWA] Forgot Password Form
- [x] Forgot Password form
- [x] Ensure that the whole flow works

Screenshot 2023-04-12 at 09 38 16

### PR DESCRIPTION
**Pull request by Kronnox**
Fix #208 

Test URLs:
- Before: https://main--maidenform--hlxsites.hlx.live/
- After: https://issue-208--maidenform--hlxsites.hlx.live/

Forgot Password Page:
- Original: https://www.maidenform.com/customer/account/forgotpassword/
- Franklin: https://issue-208--maidenform--hlxsites.hlx.live/customer/account/forgotpassword

Change Password Page:
- Original: https://www.maidenform.com/customer/account/createpassword?id=0&token=test (benötigt valides token)
- Franklin: https://issue-208--maidenform--hlxsites.hlx.live/customer/account/createpassword?id=0&token=test

### CODE REVIEW

**Comment by herzog31** [comment]
> But my question: I tried using the form but didn't get an email. I guess its disabled on QA...? @herzog31 I guess what's still open is the implementation for the page where the user can actually input the password (ie. the link that is sent by email) - how/where do we implement that?

This should be part of this ticket. The link is configured in Magento. Once we have it implemented, we need to document the link for the Maidenform team in a GitHub issue and assign it to `Hanes`, so they can change it on Magento side.

**Comment by Jordaneisenburger** [comment]
You might want to remove files like `textInput.module.css`, `inputMessage.module.css` and `field.module.css` and refer to the ones in `src/pages/account/AddressBookPage/` to prevent file duplication and styling discrepancies.


Ideally these should be all colocated to their respective components instead of being loaded from `AddressBookPage` but that's a framework choise ya'll need to make.

**Comment by hannessolo** [comment]
> to prevent file duplication and styling discrepancies

@Jordaneisenburger I'm not sure about this case, but in some cases (eg for some buttons and inputs) this duplication is deliberate. That allows us to style buttons differently depending on where they are, since the current madienform styling is not particularly consistent. The pattern is to create a new css file for the button, import it in the parent, and pass the classes to the button through props.

**Comment by Kronnox** [comment]
> Ideally these should be all colocated to their respective components

@Jordaneisenburger Yes, I agree. This was how I had it originally. Since this wasn't exclusive to this issue I decided to go with consistency and copy the files instead (like it's done everywhere else). IMO this needs to be adressed separately, as this includes all sorts of components and pages.

> That allows us to style buttons differently depending on where they are

@hannessolo Still, I do think this could be improved. There is definitely code duplication going on here right now. The pages should only contain the variations of the different component styling. As of now, styling is mostly done in the pages where it could be moved to the components.

Adding a separate issue to refactor / clean this up (after all of the ongoing changes in other branches using these components have been merged together) would be probably the best.

File: src/pages/create-password/resetPasswordPage.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+ResetPasswordPage.propTypes = {
+    classes: shape({
+        root: string,
+        header: string,
+        contentContainer: string
+    })
+};
+
+export default ForgotPasswordPage;
```
Comment (herzog31) [inline src/pages/create-password/resetPasswordPage.js:34]:
```suggestion
export default ResetPasswordPage;
```

**Review by herzog31 — APPROVED** [review]
LGTM 👍 
Please fix the wrong export and then you can merge.

**Pull request merged**